src/Cognito.re

```diff
@@ -201,6 +206,158 @@ let signUp =
           | _ => `CognitoUnknownError(msg)
           };
         Future.make(resolve => resolve(Belt.Result.Error(err)));
-      }
-    );
+      };
+    });
+};
+type confirmSignUpErrors = [ | `CognitoConfirmationCodeValidation(string)];
+let confirmSignUp = (config, ~username, ~confirmationCode, ()) => {
+  let params = Js.Dict.empty();
+
+  Js.Dict.set(params, "Username", Js.Json.string(username));
+  Js.Dict.set(params, "ConfirmationCode", Js.Json.string(confirmationCode));
+
+  Client.request(config, ConfirmSignUp, params)
+  ->Future.flatMapOk(res => {
+      /*
+             This is the raw response. Not sure what to do with this if anything. What is `tag:1`
+             Raw Response [ [ 200, tag: 1 ], {} ]
+       Raw Response [ [ 200, tag: 1 ], {} ] */
+      Js.log2("Raw Response", res);
```

Comment:
Delete this and the comment above.  

The response is a Record type, see ~ line 50 above for the definition.  That record field is a variant that carries a payload. The tag corresponds to a variant name, the int to the payload.  So here, this is the compiled version of:
```reason
type status = Success(int)
              ^ Tag1?  ^ Response code 200
```

Reply:
Just to get the habits right, how do you pro devs handle comments in your code when you are working with someone. Apparently, the way I talk to myself in comments doesn't work if anyone is looking at it. 

Thanks for the answer there.

Reply:
Ahh, there really are no 100% rules here.  Feel free to keep doing this in the future, I'm happy to answer the questions posed, but I like to keep the codebase clean of this sort of thing before merger.


Personally I tend to have a lot of comments to myself as I work through a new problem, then delete most of them as I figure out the approach.  The only ones I leave are explanatory, especially ones that explain _why_ I made a choice to do something a certain way.  Or to document as assumption I'm making that's not explicit.  I don't think there's a right answer??

Here, I (hope I) answered your question, so I figured let's kill the comment?